compiler/util/files.cpp

```diff
@@ -91,7 +91,23 @@ static void addPath(const char* pathVar, std::vector<const char*>* pathvec) {
       colon++;                            // and advance to the next
     }
 
-    pathvec->push_back(astr(dirString));
+    // FIXME (Maybe?)
+    // Following the precedent of $PATH on Unix, we should
+    // treat empty strings between colons like  :: or trailing/leading
+    // colons as meaning to add the current directory to the path.
+    // If we don't include the current directory in the CHPL_LIB_PATH by
+    // default, this behavior below is incorrect, and instead of ignoring
+    // empty strings, it should figure out the current directory and add
+    // that to the path.
+    // Alternatively, we can alter the compiler to throw -L . when
+    // CHPL_LIB_PATH has empty strings in between colons.
+    // However, if we do include the current directory in CHPL_LIB_PATH
+    // by default, then this doesn't need fixing, delete this FIXME.
+
+    // ignore empty strings
+    if (dirString && *dirString) {
```

Comment:
I have a slight preference for `dirString && strlen(dirString) == 0` to make the code more self documenting. But I am fine with what you have, too.

Reply:
I will change the conditional to `dirString && strlen(dirString) > 0`

Reply:
Oh or maybe you meant I should change it to
```
    if (!dirString || strlen(dirString) == 0) {
      continue;
    }
    pathvec->push_back(astr(dirString));
```
?

Reply:
Oh, yeah sorry, I meant the first one. But no need to fix it after merging it, it is completely fine.